src/wallet/gtest/test_wallet_zkeys.cpp

```diff
@@ -353,9 +351,7 @@ TEST(wallet_zkeys_tests, write_cryptedzkey_direct_to_db) {
     ASSERT_EQ(0, addrs.size());
 
     // Add random key to the wallet
-    auto address = wallet.GenerateNewZKey();
-    ASSERT_NE(boost::get<libzcash::SproutPaymentAddress>(&address), nullptr);
-    auto paymentAddress = boost::get<libzcash::SproutPaymentAddress>(address);
+    auto paymentAddress = wallet.GenerateNewSproutZKey();
```

Comment:
Shouldn't we still have the null assertion? The name change in this function does not ensure this check is done automatically for us.

Reply:
see below